contracts/contracts/buyback/BaseBuyback.sol

```diff
@@ -135,110 +153,135 @@ abstract contract BaseBuyback is Initializable, Strategizable {
+        // slither-disable-next-line unused-return
+        _updateBuybackSplits();
+    }
+
+    function _swapToken(
+        address tokenOut,
         uint256 oTokenAmount,
-        uint256 minOGV,
-        uint256 minCVX
-    ) external onlyGovernorOrStrategist nonReentrant {
+        uint256 minAmountOut,
+        bytes calldata swapData
+    ) internal returns (uint256 amountOut) {
         require(oTokenAmount > 0, "Invalid Swap Amount");
-        require(universalRouter != address(0), "Uniswap Router not set");
-        require(rewardsSource != address(0), "RewardsSource contract not set");
-        require(minOGV > 0, "Invalid minAmount for OGV");
-        require(minCVX > 0, "Invalid minAmount for CVX");
+        require(swapRouter != address(0), "Swap Router not set");
+        require(minAmountOut > 0, "Invalid minAmount");
 
-        uint256 ogvBalanceBefore = IERC20(ogv).balanceOf(rewardsSource);
-        uint256 cvxBalanceBefore = IERC20(cvx).balanceOf(address(this));
+        // Transfer OToken to Swapper for swapping
+        // slither-disable-next-line unchecked-transfer unused-return
+        IERC20(oToken).transfer(swapRouter, oTokenAmount);
 
-        uint256 amountInForOGV = oTokenAmount / 2;
-        uint256 amountInForCVX = oTokenAmount - amountInForOGV;
+        // Swap
+        amountOut = ISwapper(swapRouter).swap(
+            oToken,
+            tokenOut,
+            oTokenAmount,
+            minAmountOut,
+            swapData
+        );
 
-        // Build swap input
-        bytes[] memory inputs = new bytes[](2);
+        require(amountOut >= minAmountOut, "Higher Slippage");
 
-        inputs[0] = abi.encode(
-            // Send swapped OGV directly to RewardsSource contract
-            rewardsSource,
-            amountInForOGV,
-            minOGV,
-            _getSwapPath(ogv),
-            false
-        );
+        emit OTokenBuyback(oToken, tokenOut, minAmountOut, amountOut);
+    }
 
-        inputs[1] = abi.encode(
-            // Buyback contract receives the CVX to lock it on
-            // behalf of Strategist after the swap
-            address(this),
-            amountInForCVX,
-            minCVX,
-            _getSwapPath(cvx),
-            false
-        );
+    /**
+     * @dev Swaps `oTokenAmount` to OGV
+     * @param oTokenAmount Amount of OUSD/OETH to swap
+     * @param minOGV Minimum OGV to receive for oTokenAmount
+     * @param swapData 1inch Swap Data
+     */
+    function swapForOGV(
+        uint256 oTokenAmount,
+        uint256 minOGV,
+        bytes calldata swapData
+    ) external onlyGovernorOrStrategist nonReentrant {
+        (uint256 _ogvAmount, ) = _updateBuybackSplits();
+        require(_ogvAmount >= oTokenAmount, "Balance underflow");
+        require(rewardsSource != address(0), "RewardsSource contract not set");
+
+        // Subtract the amount to swap from net balance
+        ogvShare = _ogvAmount - oTokenAmount;
```

Comment:
This could use unsafe math since a check is made on line 227. You could also remove the require check,

Reply:
This is done